### PR TITLE
Pytest, Coverage, Lint

### PR DESCRIPTION
**Pull request by paulolimac**
Addresses #5 .

Implements some unit tests to the code.
Uses code-test-coverage.
Checks some typos.

For more info, see commits.

### CODE REVIEW

**Comment by fkztw** [comment]
@paulolimac Thanks for your great pull request, I will review it ASAP.
I am also amazing about your git commit messages.
May I ask a question?
Are you using some IDE or tool to help you generate those great commit messages, or just using a `.gitmessage` template and type them manually?
Again, thanks for your great contribution!

**Comment by paulolimac** [comment]
**tldr**:

Parts are done manually, and parts in automatic way.
It can be done with scripts (bash, python, ruby, you choice).
Github have a lot of repos with this scripts.

---

Hi @M157q

I don't use IDEs (i don't like it).
I only use Gedit, Nano, Git, Python, Terminal command line and Firefox.

My commit messages are prepared in manual and automatic ways.
It is not my focus for now to do automagically it all.
Only in a far far future.

My focus is in (re)write the messages.
I desire to obtain them in a good fit for my purposes.

But it is very easy to automate this.
It is only 2 things:

1. a bunch of scripts
2. that results in outputed text in a file. :smile:

For you, some references:

- gitmessage template (you mention it)
  - general-base [template](http://codeinthehole.com/tips/a-useful-template-for-commit-messages/)
  - look for gists like [this](https://gist.github.com/RichardBronosky/beacdea6106de0d0c7d36fbaf625139e) and [that](https://gist.github.com/adeekshith/cd4c95a064977cdc6c50)
  - learn from big shoulders like [this](https://github.com/git/git/blob/master/Documentation/SubmittingPatches) and [that](https://github.com/torvalds/linux/blob/master/Documentation/process/submitting-patches.rst)
- gitmessage templateS (create specific templateS)
  - test template
  - documentation template
  - feature template
  - fix template
  - chore template
  - and so on...
  - when commit, choose the specific template: `git commit --template=<file>`
- [git hooks](https://git-scm.com/docs/githooks)
  - [in your taste](https://git-scm.com/book/gr/v2/Customizing-Git-Git-Hooks)
  - [in your language](https://www.atlassian.com/git/tutorials/git-hooks)
  - [don't redo the wheel](https://github.com/git-hooks/git-hooks)
- github [webhooks](https://developer.github.com/v3/repos/hooks/)
  - [pull-requests](https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/) and [issues](https://help.github.com/articles/creating-an-issue-template-for-your-repository/) accept [templates for itself too](https://github.com/devspace/awesome-github-templates).
- python modules to execute git (or wrapped it, binding, faked...):
  - [duwich](https://github.com/jelmer/dulwich)
  - [GitPython](https://github.com/gitpython-developers/GitPython)
  - [pygit](https://github.com/benhoyt/pygit)
  - [pygit2](https://github.com/libgit2/pygit2)
  - [gittle](https://github.com/FriendCode/gittle)
  - write your own script
  - etc
  - call [shell commands](https://github.com/amoffat/sh) with `import sh`
- use bash scripts, third-part hooks etc

That is all. :octocat:

**Review by fkztw — APPROVED** [review]

**Comment by fkztw** [comment]
ref: #5
